.pipelines/templates/e2e-upgrade-template.yml

```diff
@@ -0,0 +1,94 @@
+jobs:
+  - job: e2e_upgrade_tests
+    timeoutInMinutes: 10
+    cancelTimeoutInMinutes: 5
+    workspace:
+      clean: all
+    variables:
+      - name: REGISTRY_NAME
+        value: kind-registry
+      - name: REGISTRY_PORT
+        value: 5000
+      - name: KUBERNETES_VERSION
+        value: v1.21.1
+      - name: KIND_CLUSTER_NAME
+        value: kms
+      - name: KIND_NETWORK
+        value: kind
+      - group: kubernetes-kms
+
+    steps:
+      - task: GoTool@0
+        inputs:
+          version: 1.16
+
+      - script: make e2e-install-prerequisites
+        displayName: "Install e2e test prerequisites"
+
+      - script: |
+          . scripts/setup-local-registry.sh
+        displayName: "Setup local registry"
+        env:
+          REGISTRY_NAME: $(REGISTRY_NAME)
+          REGISTRY_PORT: $(REGISTRY_PORT)
+
+      - script: |
+          version=$(git tag -l --sort=v:refname | tail -n 1)
+          echo "##vso[task.setvariable variable=LATEST_KMS_VERSION]$version"
+
+          echo "Latest released kms version - $version"
+        displayName: "Get latest released version"
+      
+      - template: manifest-template.yml
+        parameters:
+          registry: mcr.microsoft.com/oss/azure/kms
+          imageName: keyvault
+          imageVersion: $(LATEST_KMS_VERSION)
+
+      - script: |
+          . scripts/setup-kind-cluster.sh &
+          . scripts/connect-registry.sh &
+          wait
+        displayName: "Setup kind cluster with azure kms plugin"
+        env:
+          REGISTRY_NAME: $(REGISTRY_NAME)
+          REGISTRY_PORT: $(REGISTRY_PORT)
+          KUBERNETES_VERSION: $(KUBERNETES_VERSION)
+          KIND_CLUSTER_NAME: $(KIND_CLUSTER_NAME)
+          KIND_NETWORK: $(KIND_NETWORK)
+
+      - template: cluster-health-template.yml
+      - template: kind-debug-template.yml
+
+      - script: make e2e-test
+        displayName: "Run e2e tests"
+
+      - script: |
+          echo "##vso[task.setvariable variable=LOCAL_IMAGE_VERSION]$(git rev-parse --short HEAD)"
+        displayName: "Update Image version"
+      
+      - template: manifest-template.yml
+        parameters:
+          registry: localhost:$(REGISTRY_PORT)
+          imageName: keyvault
+          imageVersion: e2e-$(LOCAL_IMAGE_VERSION)
+
```

Comment:
Let's add a comment here on how the kms plugin is being upgraded.  The path is mounted in kind cluster, any changes in the host will automatically be reflected in `/etc/kubernetes/manifests` mount path.

Reply:
done